### PR TITLE
Map optimizer and loss strings to PyTorch objects

### PR DESCRIPTION
**Pull request by vloncar**
Addresses #22. New runtime parameter is added `--worker-optimizer-params` to add parameters to the optimizer, in the string representation of a dict. E.g., `--worker-optimizer sgd --worker-optimizer-params "{'lr':0.1, 'momentum': 1, 'nesterov': True}"`

### CODE REVIEW

**Comment by vloncar** [comment]
Now also includes mapping of loss string to pytorch objects (#21)

**Comment by vlimant** [comment]
@vloncar the mapping is a nice feature, a practical mapping would be "categorical_crossentropy" => the same loss in keras and torch

**Comment by vloncar** [comment]
I was thinking about that, but then maybe someone who is interested only in torch wouldn't want to write loss as it is named in Keras. How about we support both, i.e., accept the practical mappings, but also accept the class name?

**Comment by vlimant** [comment]
yes to supporting both indeed !

**Comment by vloncar** [comment]
Our PyTorch expert @thongonary points out a problem with the idea of practical mapping.

> Keras by default takes softmax as the input for CategoricalCrossEntropy, while Pytorch's CrossEntropy will do LogSoftmax to the inputs and then compute the loss. It has to implemented differently from the model. So CategoricalCrossEntropy in Keras (with softmax as the last activation function in the model) would be equivalent to Pytorch's CrossEntropy (with no activation in the last layer) or NLLLoss (with LogSoftmax as the activation of the last layer).

**Comment by vlimant** [comment]
right, which means one has to be careful anyways. whichever is the example already implemented, the MPIDriver.py command needs to stay the same except for --torch / --tf

**Comment by vlimant** [comment]
sweet !
